mne/datasets/utils.py

```diff
@@ -314,32 +340,33 @@ def _data_path(path=None, force_update=False, update_path=True, download=True,
         version=None,   # Data versioning is decoupled from MNE-Python version.
         registry=None,  # Registry is loaded from file, below.
         urls=pooch_urls,
-        retry_if_failed=2  # 2 retries = 3 total attempts
+        retry_if_failed=2  # 2 retries = 3 total attempts,
     )
-    # load the checksum registry
-    registry = _resource_path('mne.data', 'dataset_checksums.txt')
-    fetcher.load_registry(registry)
-
-    # update the keys that are versioned
-    versioned_keys = {
-        f'{TESTING_VERSIONED}.tar.gz': fetcher.registry['mne-testing-data'],
-        f'{MISC_VERSIONED}.tar.gz': fetcher.registry['mne-misc-data']}
-    fetcher.registry.update(versioned_keys)
-    for key in ('testing', 'misc'):
-        del fetcher.registry[f'mne-{key}-data']
+
+    # create temporary checksum registry
+    with tempfile.TemporaryDirectory() as tmpdir:
+        registry = op.join(tmpdir, 'dataset_checksums.txt')
+
+        # write the md5 hashes
+        with open(registry, 'w') as fout:
+            for archive_name, dataset_hash in pooch_hash_mapping.items():
+                # write to the checksums file registry
+                fout.write(f'{archive_name} {dataset_hash}\n')
+
+        # load the checksum registry
+        fetcher.load_registry(registry)
```

Comment:
this is actually not necessary.  The registry can be passed as a `dict` when the fetcher is instantiated: `fetcher = pooch.create(..., registry=pooch_hash_mapping)`

Reply:
Ah nice, that's great.